packages/core/components/RichTextEditor/components/InlineMenu/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, ReactNode, useRef } from "react";
+import { MenuBar } from "../MenuBar";
+import { RichTextMenuItem, RichTextSelector } from "../../types";
+import { ActionBar } from "../../../ActionBar";
+import { useAppStoreApi } from "../../../../store";
+import { useActiveEditor } from "../../context";
+
+export function InlineMenu({
+  menuConfig,
+  selector,
+  id,
+}: {
+  menuConfig: Record<string, Record<string, RichTextMenuItem>>;
+  selector?: RichTextSelector;
+  id?: string;
+}) {
+  const appStoreApi = useAppStoreApi();
+  const { activeEditor: editor, currentInlineId } = useActiveEditor();
+
+  const defaultOverrides = useRef(appStoreApi.getState().overrides);
+  const previous = defaultOverrides.current?.actionBar;
+
+  useEffect(() => {
+    if (currentInlineId === null) {
+      appStoreApi.setState((s) => ({
+        overrides: {
+          ...defaultOverrides.current,
+        },
+      }));
+      return;
+    }
+
+    const customActionBar = ({
+      children,
+      label,
+      parentAction,
+    }: {
+      label?: string;
+      children: ReactNode;
+      parentAction: ReactNode;
+    }) => (
+      <ActionBar label={label}>
+        <ActionBar.Group>
+          <MenuBar
+            menuConfig={menuConfig}
+            editor={editor}
+            selector={selector}
+            inline
+          />
```

Comment:
How easy is it for the user to implement these controls in their own action bar?